airflow/configuration.py

```diff
@@ -403,6 +403,64 @@ def inversed_deprecated_sections(self):
     upgraded_values: dict[tuple[str, str], str]
     """Mapping of (section,option) to the old value that was upgraded"""
 
+    legacy_incompatible_defaults: dict[tuple[str, str], tuple[Pattern, str, str]] = {
+        ("logging", "log_filename_template"): (
+            re.compile(
+                r"^(?:" + re.escape("{{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log") + r"|"
+                r"dag_id=\{\s*ti\.dag_id\s*\}/run_id=\{\s*ti\.run_id\s*\}/task_id=\{\s*ti\.task_id\s*\}/"
+                r"\{%%\s*if\s+ti\.map_index\s*>=\s*0\s*%%\}map_index=\{\s*ti\.map_index\s*\}/\{%%\s*endif\s*%%\}"
+                r"attempt=\{\s*try_number\s*\}\.log" + r")$"
+            ),
+            "3.0",
+            "The default value for 'log_filename_template' from Airflow 2 may break task logs in the new UI.",
+        ),
+        ("core", "dag_ignore_file_syntax"): (
+            re.compile(r"^regexp$"),
+            "3.0",
+            "The default value changed from 'regexp' in Airflow 2.x to 'glob' in Airflow 3.0.",
+        ),
+        ("api", "auth_backends"): (
+            re.compile(r"^airflow\.api\.auth\.backend\.session$"),
+            "3.0",
+            "The default auth backend has changed from 'airflow.api.auth.backend.session' to "
+            "'airflow.providers.fab.auth_manager.api.auth.backend.session' in Airflow 3.0.",
+        ),
+        ("scheduler", "task_instance_heartbeat_timeout"): (
+            re.compile(r"^300$"),
+            "3.0",
+            "Airflow 2.x used 'scheduler_zombie_task_threshold' with a default of 300. In 3.0 this setting "
+            "has been renamed to 'task_instance_heartbeat_timeout' (with the same numeric default), which may "
+            "cause legacy configurations to be ignored.",
+        ),
+    }
+
+    def handle_incompatible_airflow2_defaults(self, upgrade: bool | None = True) -> None:
```

Comment:
```suggestion
    def handle_incompatible_airflow2_defaults(self, upgrade: bool) -> None:
```

Should always be a bool.

Reply:
changed it